File: PostsSPA/src/components/spa/post/models/post.ts
import { Poster } from './poster';
import { Comment } from '../../comment/models/comment';

export class Post {
  Id: number;
  Title: string;
  Body: string;
  Poster: Poster;
  Comments: Comment[];

  constructor(params: Post = {} as Post) {
    const {
      Id = null,
      Title = null,
      Body = null,
      Poster = null,
      Comments = []
    } = params;

    this.Id = Id;
    this.Title = Title;
    this.Body = Body;
    this.Poster = Poster;
    this.Comments = Comments;
  }
}

File: PostsAPI/Models/Post.cs
using System.Collections.Generic;

namespace PostsAPI.Models
{
    public class Post
    {
        public Poster Poster { get; set;}

        public int Id { get; set; }

        public string Title { get; set; }

        public string Body { get; set; }

        public IEnumerable<Comment> Comments { get; set; }
    }
}